File: Daily/Binary_Gap/binary__gap.py
from typing import Optional, List
import heapq
from collections import defaultdict


class Solution:
    def binaryGap(self, n: int) -> int:
        found = 0
        count = 0
        started = False
        while n:
            tmp = n % 2
            if tmp and not started:
                count += 1
                started = True
            elif tmp and started:
                found = max(found, count)
                count = 1
            elif started:
                count += 1
            n >>= 1
        return found


if __name__ == "__main__":
    sol = Solution()
    print(sol.binaryGap(6))
    print("Running Solution...")

File: Daily/Binary_Gap/Binary_Gap.cpp
// Binary_Gap.cpp
//
// Compile locally with:
// clang++ Binary_Gap.cpp -std=c++17 -Wall -Wextra -O2 -o run && ./run
//
// NOTE:
// - Remove main() before submitting to LeetCode.
// - Keep only the class Solution definition.
//

#include <algorithm>
#include <array>
#include <cassert>
#include <cctype>
#include <climits>
#include <cmath>
#include <deque>
#include <functional>
#include <iostream>
#include <map>
#include <numeric>
#include <queue>
#include <set>
#include <stack>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>

using namespace std;

class Solution
{
public:
    int binaryGap(int n)
    {
        int count = 0;
        int tmp = 0;
        bool found = false;
        int step = 0;
        for (int i = 0; i < 32; i++)
        {
            tmp = n % 2;
            if (tmp == 1 && !found)
            {
                found = true;
                step++;
            }
            else if (found && tmp == 1)
            {
                count = max(count, step);
                step = 1;
            }
            else if (found)
            {
                step++;
            }
            n >>= 1;
        }

        return count;
    }
};

int main()
{
    cout << "Running Binary_Gap.cpp..." << endl;

    Solution sol;

    // TODO:
    // Add local test calls here

    return 0;
}
